code/modules/projectiles/gun.dm

```diff
@@ -170,7 +170,22 @@
 		user << "<span class='userdanger'>You need both hands free to fire [src]!</span>"
 		return
 
-	process_fire(target,user,1,params)
+	//DUAL (or more!) WIELDING
+	var/bonus_spread = 0
+	var/loop_counter = 0
+	if(ishuman(user) && user.a_intent == "harm")
+		var/mob/living/carbon/human/H = user
+		for(var/obj/item/weapon/gun/G in H.held_items)
+			if(G == src)
+				continue
+			else if(G.can_trigger_gun(user))
+				bonus_spread += 24 * G.weapon_weight
+				loop_counter++
+				spawn(loop_counter)
+					G.recharge_newshot()
```

Comment:
I honestly don't exactly remember since I wrote this months ago, but I do remember guns wouldn't process the chamber/reload properly when you were forcing them to fire offhand rather than going through all the steps of an actual click/attack, so forcing it to reload was needed.